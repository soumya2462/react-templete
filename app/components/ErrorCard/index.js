/**
 *
 * ErrorCard
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import If from '@components/If';
import Card from '@components/Card';
import { IntlShape } from '@utils';
import { T } from '@components/T';

const ErrorCard = ({ error, intl }) => {
  return (
    <Card data-testid="error-card" color={error ? 'red' : 'grey'} title={intl.formatMessage({ id: 'repo_list' })}>
      <If condition={error} otherwise={<T data-testid="default-message" id={error || 'repo_search_default'} />}>
        <T data-testid="error-message" text={error} />
      </If>
    </Card>
  );
};

ErrorCard.propTypes = { error: PropTypes.string, intl: IntlShape };

export default injectIntl(ErrorCard);
