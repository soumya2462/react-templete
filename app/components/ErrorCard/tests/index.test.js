/**
 *
 * Tests for ErrorCard
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import ErrorCard from '../index';

describe('<ErrorCard />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<ErrorCard />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 ErrorCard component', () => {
    const { getAllByTestId } = renderWithIntl(<ErrorCard />);
    expect(getAllByTestId('error-card').length).toBe(1);
  });
});
