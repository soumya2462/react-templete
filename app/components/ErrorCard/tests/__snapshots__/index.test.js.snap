// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorCard /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-card ant-card-bordered Card__CustomCard-sc-7y2i64-0 gYsQUy"
      color="grey"
      data-testid="error-card"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            Repository List
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <p
          class="T__StyledText-gjlic1-0 kYcyRq"
          data-testid="default-message"
        >
          Search for a repository by entering it's name in the search box
        </p>
      </div>
    </div>
  </div>
</body>
`;
